src/prng.c

```diff
@@ -0,0 +1,92 @@
+/******************************************************************************
+ *
+ * Copyright 2017 Xaptum, Inc.
+ * 
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ * 
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License
+ *
+ *****************************************************************************/
+
+#include <ecdaa/prng.h>
+
+#ifndef DISABLE_LIBSODIUM_RNG_SEED_FUNCTION
+
+#include <sodium.h>
+
+#if defined(__linux__)
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/ioctl.h>
+#include <linux/random.h>
+#endif  // defined(__linux__)
+
+int ecdaa_prng_init(struct ecdaa_prng *prng_in)
```

Comment:
This implementation should call `ecdaa_prng_init_custom` internally, to avoid code duplication.  Generate the seed and call `ecdaa_prng_init_custom` with it.

Reply:
I think this hinges on the signature of `ecdaa_prng_init_custom`, as per above.

Reply:
:+1:

Reply:
Even if we keep the function arg for `ecdaa_prng_init_custom`, `ecdaa_prng_init` should call through to `ecdaa_prng_init_custom`. After all, we're just providing a default libsodium-based variant.  If using `ecdaa_prng_init_custom` for that is too difficult for us, then what hope does an end user have?  It would imply that the `ecdaa_prng_init_custom` API needs more work.

Reply:
Agreed. Implementation has been updated now.